### PR TITLE
fix nms numeric stable bug

### PR DESCRIPTION
**Pull request by fsx950223**
I fix the nms bug on float16 kernel, I have uploaded the code on https://github.com/fsx950223/tensorflow/commits/non_max_suppression.
Here is my minimal example:
```
import tensorflow as tf
boxes_f = open('/tensorflow_src/tensorflow/examples/tests/boxes.tensor', 'rb').read()
scores_f = open('/tensorflow_src/tensorflow/examples/tests/scores.tensor', 'rb').read()
boxes = tf.io.parse_tensor(boxes_f, tf.float16)
scores = tf.io.parse_tensor(scores_f, tf.float16)
nms_top_idx, nms_scores, nms_valid_lens = tf.raw_ops.NonMaxSuppressionV5(
    boxes=boxes,
    scores=scores,
    max_output_size=100,
    iou_threshold=1.0,
    score_threshold=0.3,
    soft_nms_sigma=(0.5 / 2),
    pad_to_max_output_size=True)
tf.print(nms_top_idx, nms_scores, nms_valid_lens)
```
You can download tensors from [boxes](https://drive.google.com/file/d/1vSM0-Vy8ivG0PakERWLvJinhDm41zhco/view?usp=sharing), [scores](https://drive.google.com/file/d/1qbVxRFp_jSTKY_zYvvzvpIkJJ42Auq9S/view?usp=sharing).

Could you review it and test internals?

### CODE REVIEW

**Comment by mingxingtan** [comment]
@fsx950223 could you prepare a single PR  for your branch (so I can patch it and give it a try)?

File: efficientdet/keras/postprocess.py [hunk 1]
```diff
@@ -220,8 +221,8 @@ def postprocess_combined(params, cls_outputs, box_outputs, image_scales=None):
   Returns:
     A tuple of batch level (boxes, scores, classess, valid_len) after nms.
   """
-  cls_outputs = to_list(cls_outputs)
-  box_outputs = to_list(box_outputs)
+  cls_outputs = util_keras.fp16_to_fp32_nested(to_list(cls_outputs))
+  box_outputs = util_keras.fp16_to_fp32_nested(to_list(box_outputs))
```
Comment (mingxingtan) [inline efficientdet/keras/postprocess.py:225]: Is this necessary?

Comment (fsx950223) [reply efficientdet/keras/postprocess.py:225]: Combined nms only has float32 kernel

File: efficientdet/keras/util_keras.py [hunk 2]
```diff
@@ -146,10 +146,20 @@ def restore_ckpt(model,
...
-  if (tf.train.list_variables(ckpt_path_or_file)[0][0] ==
+  var_list = tf.train.list_variables(ckpt_path_or_file)
+  if (var_list[0][0] ==
       '_CHECKPOINTABLE_OBJECT_GRAPH'):
     try:
-      model.load_weights(ckpt_path_or_file)
+      # Use custom checkpoint solves mismatch shape issue.
+      keys = {var[0].split('/')[0] for var in var_list}
+      keys.discard('_CHECKPOINTABLE_OBJECT_GRAPH')
+      if skip_mismatch:
```
Comment (mingxingtan) [inline efficientdet/keras/util_keras.py:156]: "skip_mismatch" should cover all variables that have shape mismatched? (not only class_net)?

**Review by mingxingtan — APPROVED** [review]

File: efficientdet/keras/util_keras.py (hunk 2 above)
Comment (fsx950223) [reply efficientdet/keras/util_keras.py:156]: The current Checkpoint approach doesn't support shape mismatch. Maybe I should add exclude vars back?

**Review by mingxingtan — APPROVED** [review]
Thanks @fsx950223

**Pull request merged**